src/executor/insert.rs

```diff
@@ -83,15 +83,21 @@ where
     A: ActiveModelTrait,
 {
     // TODO: Postgres instead use query_one + returning clause
-    let result = match db {
+    let last_insert_id = match db {
         #[cfg(feature = "sqlx-postgres")]
         DatabaseConnection::SqlxPostgresPoolConnection(conn) => {
             let res = conn.query_one(statement).await?.unwrap();
-            crate::query_result_into_exec_result(res)?
+            res.try_get("", "last_insert_id").unwrap_or_default()
         }
-        _ => db.execute(statement).await?,
+        _ => {
+            db.execute(statement).await?
+            .last_insert_id()
+            .to_string()
+            .parse()
```

Comment:
Yeah this looks much more sensible. May I also know why is it necessary to `to_string` and `parse` again?

Reply:
Because we cannot convert e.g. `u64` to `i32` directly

Reply:
how about this https://stackoverflow.com/questions/28273169/how-do-i-convert-between-numeric-types-safely-and-idiomatically?

Reply:
I saw this StackOverflow this morning. I don't think it helps :(

Reply:
Why not?

Reply:
We can't use `TryFrom` or `From` for it

Reply:
Why not?

Reply:
If we want `TryFrom` or `From` `u32` to `ValueType`, this requires `ValueType: From<u32>`. Which is not reasonable

Reply:
But having to stringify and parse is not much more sensible either

Reply:
Agree